### ISSUE TITLE
[Mailer] google-mailer DSN configuration not valid anymore

### ISSUE THREAD

**Issue by jeremyjaouen** [symfony/symfony#46826]
### Symfony version(s) affected

5.4.*

### Description

With recents change on GMail, the following configuration for MAILER_DSN: 
`MAILER_DSN=gmail://USERNAME:PASSWORD@default` is no more valid, as the connection using username and password are not able anymore.

Is there any existing workaround or another configuration to be able to use SMTP with Google-Mailer ?

### How to reproduce

- Install symfony/google-mailer
- Setup your .env with `MAILER_DSN=gmail://USERNAME:PASSWORD@default` (replace your username and password with your own credentials
- Implement use the MailerInterface to send an email
- Your Request shouldn't succeed with following error:
```
message: Failed to authenticate on SMTP server with username "xxxxxxx" using the following authenticators: "LOGIN", "PLAIN", "XOAUTH2". Authenticator "LOGIN" returned "Expected response code "235" but got code "535", with message "535-5.7.8 Username and Password not accepted. Learn more at
535 5.7.8  https://support.google.com/mail/?p=BadCredentials y18-20020a05622a121200b002f90a33c78csm17479301qtx.67 - gsmtp".". Authenticator "PLAIN" returned "Expected response code "235" but got code "535", with message "535-5.7.8 Username and Password not accepted. Learn more at
535 5.7.8  https://support.google.com/mail/?p=BadCredentials y18-20020a05622a121200b002f90a33c78csm17479301qtx.67 - gsmtp".". Authenticator "XOAUTH2" returned "Expected response code "235" but got code "334", with message "334 eyJzdGF0dXMiOiI0MDAiLCJzY2hlbWVzIjoiQmVhcmVyIiwic2NvcGUiOiJodHRwczovL21haWwuZ29vZ2xlLmNvbS8ifQ==".".
```

**Comment by 1ed** [comment]
As stated in the linked documentation you should enable 2-fa and use an App Password (https://support.google.com/accounts/answer/185833) or oauth to sign in. Using the regular password and less secure app option is not supported any more https://support.google.com/accounts/answer/6010255

> To help keep your account secure, from May 30, 2022, ​​Google no longer supports the use of third-party apps or devices which ask you to sign in to your Google Account using only your username and password.

**Comment by jeremyjaouen** [comment]
Indeed, I have already read the doc from Google, but it remains that the Symfony Mailer Documentation is no more up to date regarding SMTP and Gmail

**Comment by fabpot** [comment]
@jeremyjaouen Can you submit a PR to update the docs?
